### PR TITLE
Add tests for some commands with 'print-xml' option

### PR DESCRIPTION
**Pull request by rbian**

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_change_media.py [hunk 1]
```diff
@@ -228,13 +228,23 @@ def update_device(vm_name, init_iso, options, start_vm):
...
-                vm.destroy()
-            if vm.is_dead():
-                vm.start()
-            session = vm.wait_for_login()
-            check_media(session, check_file, action, rw_floppy_test)
-            session.close()
+            if 'print-xml' in options:
+                vmxml_new = vm_xml.VMXML.new_from_dumpxml(vm_name)
+                if source in vmxml_new:
+                    raise error.TestFail("See device with 'print-xml' option")
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_change_media.py]: So the purpose here is make sure '--print-xml' option will change media actually, right? Maybe you should describe the error more accurate.

File: libvirt/tests/src/virsh_cmd/domain/virsh_change_media.py [hunk 2]
```diff
@@ -228,13 +228,23 @@ def update_device(vm_name, init_iso, options, start_vm):
...
+            else:
+                if options == "--config" and vm.is_alive():
+                    vm.destroy()
+                if vm.is_dead():
+                    vm.start()
+                if 'block' in options and 'eject' not in action:
+                    vmxml_new = vm_xml.VMXML.new_from_dumpxml(vm_name)
+                    logging.debug("vmxml: %s", vmxml_new)
+                    if not vm_xml.VMXML.check_disk_type(vm_name, source, "block"):
+                        raise error.TestFail("Disk type isn't correct")
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_change_media.py]: It's better to tell the expect type for easy debugging.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiflist.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def parse_interface_details(output):
                 iface_cmd = {}
         return ifaces_cmd
 
-    def check_output(output, vm):
+    def check_output(ifaces_actual, vm):
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiflist.py]: So this is a bug fix, right?

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domiflist.cfg [hunk 4]
```diff
@@ -32,6 +32,16 @@
...
+                - iface_options:
+                    domiflist_domname_options = "name"
+                    start_vm = yes
+                    attach_iface = "yes"
+                    iface_source = "default"
+                    iface_type" = "network"
+                    variants:
+                        - print_xml:
+                            iface_model = "virtio"
+                            at_suffix = "--print-xml"
```
Comment (will-Do) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_domiflist.cfg]: 'at_suffix' is not easy to understand, maybe you can find a more readable name

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiflist.py [hunk 5]
```diff
@@ -148,15 +144,31 @@ def add_iface(vm):
...
+        old_iflist = parse_interface_details(result.stdout)
+        logging.debug("old: %s", old_iflist)
+        # Attach interface for testing.
+        if attach_iface:
+            add_iface(vm, attach_suffix)
+
+        vm.verify_alive()
         result = virsh.domiflist(options, additional_options, ignore_status=True)
+        new_iflist = parse_interface_details(result.stdout)
+        logging.debug("new: %s", new_iflist)
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiflist.py]: Let's make the message more clear, such as:
"Old interface list:"
"New interface list:"

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domiflist.cfg [hunk 6]
```diff
@@ -32,6 +32,16 @@
                             iface_model = "e1000"
                         - rtl8139_iface:
                             iface_model = "rtl8139"
+                - iface_options:
+                    domiflist_domname_options = "name"
+                    start_vm = yes
+                    attach_iface = "yes"
+                    iface_source = "default"
+                    iface_type" = "network"
```
Comment (balamuruhans) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_domiflist.cfg]: iface_type" = "network" , there is a typo error.

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiflist.py (hunk 3 above)
Comment (rbian) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiflist.py]: Not bug fix, just adjust code logic.

**Comment by will-Do** [comment]
Okay, then we can merge this pr

**Pull request merged**